File: 프로그래머스/lv2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    let sum = brown + yellow;
    
    for(let 가로=3; ; 가로++){
        let 세로 = sum / 가로;
        if(세로 > 가로 || 세로%1 !== 0) continue;
        if((가로-2) * (세로-2) === yellow){
            return [가로,세로]
        }
    }
}

File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];

        int sum = brown + yellow;
        int width = 0;
        int minWidth = 3;

        for (int i = minWidth; i < brown; i++) {
            int height = sum / i;
            if (height > i) {
                continue;
            }
            if ((i * 2) + (height - 2) * 2 == brown) {
                width = i;
                break;
            } 
        }

        answer[0] = width;
        answer[1] = sum / width;

        return answer;
    }
}